Mathlib/CategoryTheory/Adjunction/Additive.lean

```diff
@@ -0,0 +1,122 @@
+/-
+Copyright (c) 2024 Sophie Morel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sophie More, Joël Riou
+-/
+import Mathlib.CategoryTheory.Preadditive.Yoneda.Basic
+
+/-!
+# Adjunctions between additive functors.
+
+This provides some results and constructions for adjunctions between functors on
+preadditive categories:
+* If one of the adjoint functors is additive, so is the other.
+* If one of the adjoint functors is additive, the equivalence `Adjunction.homEquiv` lifts to
+an additive equivalence `Adjunction.homAddEquivOfLeftAdjoint` resp.
+`Adjunction.homAddEquivOfRightAdjoint`.
+* We also give a version of this additive equivalence as an isomorphism of `preadditiveYoneda`
+functors (analogous to `Adjunction.compYonedaIso`), in
+`Adjunction.compPreadditiveYonedaIsoOfLeftAdjoint` resp.
+`Adjunction.compPreadditiveYonedaIsoOfRightAdjoint`.
+
+-/
+
+universe u₁ u₂ v₁ v₂
+
+namespace CategoryTheory
+
+namespace Adjunction
+
+open CategoryTheory Category Functor
+
+variable {C : Type u₁} {D : Type u₂} [Category.{v₁} C] [Category.{v₂} D] [Preadditive C]
+  [Preadditive D] {F : C ⥤ D} {G : D ⥤ C} (adj : F ⊣ G)
+
+include adj
+lemma right_adjoint_additive [F.Additive] : G.Additive where
+  map_add {X Y} f g := (adj.homEquiv _ _).symm.injective (by simp [homEquiv_counit])
+
+lemma left_adjoint_additive [G.Additive] : F.Additive where
+  map_add {X Y} f g := (adj.homEquiv _ _).injective (by simp [homEquiv_unit])
+
+/-- If we have an adjunction `adj : F ⊣ G` of functors between preadditive categories,
+and if `F` is additive, then the hom set equivalence upgrades to an `AddEquiv`.-/
+@[simps! (config := .lemmasOnly)]
+def homAddEquivOfLeftAdjoint [F.Additive] (X : C) (Y : D) :
```

Comment:
Once the lemmas `left/right_adjoint_additive` have been obtained, I think we may define only one `Adjunction.homAddEquiv` (assuming `F.Additive`). I would think we should declare it with `@[simps!]` (without `.lemmasOnly`), and add `homEquiv_add/sub/neg/zero`, `homEquiv_symm_add/...` lemmas.

Reply:
Done. The `homEquiv_symm` lemmas can all be proved by `simp`, so the linter might ask to remove them.

Reply:
Well, the linter is complaining about all the lemmas you requested: the `homAddEquiv_add` etc lemmas are not in `simp` normal form (would they be that useful if I put them in `simp` normal form, I'm not sure), and the `homAddEquiv_symm_add` etc lemmas can all be proved by `simp`...

Reply:
I checked and the automatically generated `homAddEquiv_apply` (and its `symm` version) are also not great. For example, `homAddEquiv_apply` proves that `(adj.homAddEquiv X Y) f = adj.unit.app X ≫ G.map f`, which seems like a bad idea since I had a lot of trouble with the similar lemma `homEquiv_apply` before someone had the great idea to remove its `@[simp]` attribute. 
I would want `homAddEquiv_apply` to just say that `adj.homAddEquiv X Y f = adj.homEquiv X Y f`, and I'm going to change it now to do that.

Reply:
Ah, yes, the `!` in `simps!` unfolded the definition of `homEquiv`... I do not know how to fix this apart from stating each lemma as you did.